spec/models/activity_spec.rb

```diff
@@ -45,4 +54,26 @@
     end
   end
 
+  describe '#agent_instance' do
+    it 'returns an instance of the agent class' do
+      expect(subject.agent_instance)
+        .to be_an_instance_of(subject.agent.constantize)
+    end
+
+    it 'returns the same instance for successive calls' do
+      expect(subject.agent_instance).to eq subject.agent_instance
```

Comment:
I'm not sure of exactly what you're trying to test here, but are you sure you want `eq`?  Are you testing whether it returns the same object or equivalent objects?  Your `it` statement makes it sound like you want the same object, in which case I believe you should use `be` instead of `eq`.